### ISSUE TITLE
About using the model in other datasets

### ISSUE THREAD

**Issue by Danessely** [gmberton/CosPlace#29]
Do I understand correctly that I can parse the panoramas to some other city and use your pre-trained models? That is, for each image, extract the embedding and save them. Then submit a new image (query) to the model, get an embedding for it and find the nearest to it in the previously saved ones (using faiss for example). Or should I train a new model on my data?

**Comment by gmberton** [comment]
Yes, that is correct, there is no need to train a new model on your data.
In fact, the paper shows that our models reaches state-of-the-art results even when it is used on images from other cities.
Only one consideration: if you are using panoramas, I suggest you extract embeddings from crops, not from the whole panorama. For example you can crop the panorama in 12 images, and pass those to the model, and obtain 12 separate embeddings

**Comment by Danessely** [comment]
Panoramas are stored as screenshots. I.e. for one point a screenshot from the panorama at 0 degrees, 90, 180, 270.
So I need to extract several embeddings per image when saving to the database, or to extract several embeddings when searching from a query image?

**Comment by gmberton** [comment]
I don't understand your questions much, could you rephrase?
Anyway, the idea is to pass to the model images that are squared, or almost squared. So if you have an image that is 500 x 1000 pixels, it is good to cut it in two and obtain two embeddings from it

**Comment by Danessely** [comment]
My images are in 1920*722 resolution. Accordingly, I need to divide them into left and right parts, obtain and save the embeddings separately? And I don't need to do that for the query image.

**Comment by Danessely** [comment]
And one more question, does it make sense to bring the images to a certain resolution and normalize or can I not do that?

**Comment by gmberton** [comment]
The best way to crop the images is to make them as similar as possible as the queries.
If your images are 1920x722 one way to proceed is resizing them to 1360x512 and then obtain 3 partially overlapping crops of 512x512.
Normalization of the image before passing it to the model is necessary, as it is done in the code

**Comment by Danessely** [comment]
Thank you so much for your answers!
